src/wallet.rs

```diff
@@ -122,4 +161,21 @@ impl Wallet {
 
         Ok(address.to_string())
     }*/
+
+    fn sync_wallet(
+        wallet: &MutexGuard<bdk::Wallet<Tree>>,
+        blockchain: &ElectrumBlockchain,
+    ) -> LipaResult<()> {
+        match wallet.sync(blockchain, SyncOptions::default()) {
+            Ok(()) => {}
+            Err(e) => {
+                return match e {
+                    bdk::Error::Electrum(e) => Err(runtime_error(ElectrumServiceUnavailable, e)),
+                    bdk::Error::Sled(e) => Err(permanent_failure(e)),
+                    _ => Err(runtime_error(GenericError, "Failed to sync the BDK wallet")),
+                }
+            }
+        };
+        Ok(())
```

Comment:
```suggestion
        if let Err(e) = wallet.sync(blockchain, SyncOptions::default()) {
            return match e {
                bdk::Error::Electrum(e) => Err(runtime_error(ElectrumServiceUnavailable, e)),
                bdk::Error::Sled(e) => Err(permanent_failure(e)),
                _ => Err(runtime_error(GenericError, "Failed to sync the BDK wallet")),
            }
        }
    };
        
    Ok(())
```

Also I wonder whether the `_`-case should be a permanent_failure. If there is an error type thrown that we (the LN devs) are not explicitly aware of that it occurs, shouldn't we be informed such that we can add it to this `match` statement?

Reply:
I see your point and was questioning how to do it myself. This error has a lot of variants ([see here](https://docs.rs/bdk/0.25.0/bdk/enum.Error.html)) and for me it's not clear for all in which methods they can be thrown or if they represent a `PermanentError` or a `RuntimeError`. 

I went with a catch-all `RuntimeError` because I think it's more likely than not that this is a runtime error here. 

There's significant room for improvement in all error handling in the lib. I would suggest doing these improvements in the future if we see a need for them, but let me know what you think.

Reply:
That's fine for me.